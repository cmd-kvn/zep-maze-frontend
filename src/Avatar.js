import React from 'react';
import {emojify} from 'react-emojione';
import { UP, DOWN, LEFT, RIGHT } from './Controller.constants';

export function Avatar({ stance }) {
    switch(stance) {
        case UP: return ( <div> emojify( ':point_up_tone3:' ) </div> );
        case DOWN: return ( <div> emojify( ':point_down_tone3:' ) </div> );
        case RIGHT: return ( <div> emojify(':point_right_tone3:') </div> );
        case LEFT: return ( <div> emojify( ':point_left_tone3:' ) </div> );
        default: return ( <div> emojify( ':levitate_tone3:' ) </div>);
    }
}