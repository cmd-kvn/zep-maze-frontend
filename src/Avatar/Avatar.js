import React from 'react';
import  Emojify from 'react-emojione';
import { connect } from 'react-redux';
import styles from './Avatar.css'

export default function Avatar() {
    
    return ( <div className={styles.avatar}> <Emojify><span>:wink:</span></Emojify> </div>);
}
