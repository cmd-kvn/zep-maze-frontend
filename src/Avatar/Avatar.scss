

:local(.avatar){
    // margin-left: -16px;
    // margin-top: -16px;
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;
    padding: auto;
    display: flex;   
    justify-content: center;
    align-items: center;
}