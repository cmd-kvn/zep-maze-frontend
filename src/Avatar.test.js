import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Avatar } from './Avatar';

describe ('Avatar Component', () => {

    it('renders a levitating man emoji by default', () => {
        const wrapper = shallow(<Avatar />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })


})