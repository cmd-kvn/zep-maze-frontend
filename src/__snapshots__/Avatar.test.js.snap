// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component renders a levitating man emoji by default 1`] = `
<div>
   emojify( ':levitate_tone3:' ) 
</div>
`;
